### PR TITLE
Remove discussion about user's compute env from knitting docs

### LINKED ISSUE #546: Reword docs to focus less on end-user's compute environment
The language in the README and docs/index.rst is overly-focused on the environment in which a user may run cutting. Rather than mentioning "infrastructure", "tpu/gpu", "hybrid", "parallelizable", let's feature the toolbox API itself and let users decide how to integrate these building blocks into their patterns

### PR DESCRIPTION
**Pull request by caleb-johnson**
Fixes #546

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,15 +29,12 @@
...
-Each tool in the CKT partitions a user's problem into quantum and classical components to enable efficient use of resources constrained by scaling limits, i.e. size of quantum processors and classical compute capability. It can assign the execution of "quantum code" to QPUs or QPU simulators and "classical code" to various heterogeneous classical resources such as CPUs, GPUs, and TPUs made available via hybrid cloud, on-prem, data centers, etc. 
-
-The toolbox enables users to run parallelized and hybrid (quantum + classical) workloads without worrying about allocating and managing underlying infrastructure.
+Circuit Knitting is the process of decomposing a larger quantum circuit into many smaller circuits, executing those circuits on a quantum processor(s), and then knitting their results into a reconstruction of the original circuit's outcome.
 
 The toolbox currently contains the following tools:
 - Circuit Cutting [[1-6]](#references)
-  
+ 
+For a more detailed discussion on circuit cutting, check out our [technical guide](https://qiskit-extensions.github.io/circuit-knitting-toolbox/circuit_cutting/explanation/index.html#overview-of-circuit-cutting).
```
Comment (caleb-johnson) [inline README.md:37]: I think we're slowly moving toward this being a circuit *cutting* repo, so I think it's fine to slip this very nice document in on our main docs pages

File: docs/index.rst [hunk 2]
```diff
@@ -12,7 +12,9 @@ The toolbox currently contains the following tools:
...
 - Circuit Cutting
 
-This project is meant to evolve rapidly and, as such, does not follow `Qiskit's deprecation policy <https://github.com/Qiskit/qiskit/blob/main/DEPRECATION.md>`_.  We may occasionally make breaking changes in order to improve the user experience.  When possible, we will keep old interfaces and mark them as deprecated, as long as they can co-exist with the new ones.  Each substantial improvement, breaking change, or deprecation will be documented in the :ref:`release notes`.
+For a more detailed discussion on circuit cutting, check out our `technical guide <https://qiskit-extensions.github.io/circuit-knitting-toolbox/circuit_cutting/explanation/index.html#overview-of-circuit-cutting>`__.
+
+This project is meant to evolve rapidly and, as such, does not follow `Qiskit's deprecation policy <https://github.com/Qiskit/qiskit/blob/main/DEPRECATION.md>`__.  We may occasionally make breaking changes in order to improve the user experience.  When possible, we will keep old interfaces and mark them as deprecated, as long as they can co-exist with the new ones.  Each substantial improvement, breaking change, or deprecation will be documented in the :ref:`release notes`.
```
Comment (caleb-johnson) [inline docs/index.rst:17]: Added an underscore to make the `__` used elsewhere in this doc. It only had one underscore before (probably fine)

Comment (garrison) [reply docs/index.rst:17]: Yeah, it's fine except when it's not and breaks Sphinx. :sweat_smile:. better to have two underscores everywhere when we think of it.

File: docs/index.rst [hunk 3]
```diff
@@ -12,7 +12,9 @@ The toolbox currently contains the following tools:
 
 - Circuit Cutting
 
-This project is meant to evolve rapidly and, as such, does not follow `Qiskit's deprecation policy <https://github.com/Qiskit/qiskit/blob/main/DEPRECATION.md>`_.  We may occasionally make breaking changes in order to improve the user experience.  When possible, we will keep old interfaces and mark them as deprecated, as long as they can co-exist with the new ones.  Each substantial improvement, breaking change, or deprecation will be documented in the :ref:`release notes`.
+For a more detailed discussion on circuit cutting, check out our `technical guide <https://qiskit-extensions.github.io/circuit-knitting-toolbox/circuit_cutting/explanation/index.html#overview-of-circuit-cutting>`__.
```
Comment (garrison) [inline docs/index.rst:15]: A relative link would make the documentation more agnostic to where it is hosted.
```suggestion
For a more detailed discussion on circuit cutting, check out our `technical guide <./circuit_cutting/explanation/index.rst#overview-of-circuit-cutting>`__.
```

**Review by garrison — DISMISSED** [review]
Thanks, this looks great.

If you accept my relative-link suggestion, I want one of us to actually test it before this gets merged.  I am happy to do this.  Other links in this form work (sphinx does the rst -> html map automatically).

**Comment by caleb-johnson** [comment]
> Thanks, this looks great.
> 
> If you accept my relative-link suggestion, I want one of us to actually test it before this gets merged. I am happy to do this. Other links in this form work (sphinx does the rst -> html map automatically).

Link's working in the CI artifact. Should be good to merge. Thanks

**Pull request merged**